bowlingGame.js

```diff
@@ -0,0 +1,69 @@
+const Frame = require('./frame');
+
+class BowlingGame {
+  constructor() {
+    this.frames = [];
+  }
+
+  play() {
+    for (let i = 1; i<=10; i++) {
+      const currentFrame = this.playFrame(i);
+      this.frames.push(currentFrame);
+      if (i < 3) {
+        continue;
+      }
+      if (this.frames.at(-1).isSpare()) {
+        this.frames.at(-2).frameScores.push(this.frames.at(-1).frameScores[0]);
+      }
+      let frameIndex = this.frames.length-2;
+      while (frameIndex > 0 && this.frames.at(frameIndex).isStrike()) {
+        this.frames.at(frameIndex).frameScores.push(this.frames.at(frameIndex+1).getFrameScoresSum());
+        frameIndex--;
+      }
```

Comment:
A little hard to follow the logic here. Deconstructing the chains into mulitple lines or into a helper method would help